arbnode/dataposter/leveldb/leveldb.go

```diff
@@ -129,11 +138,17 @@ func (s *Storage[Item]) Put(ctx context.Context, index uint64, prev *Item, new *
 	if err := b.Put(key, newEnc); err != nil {
 		return fmt.Errorf("updating value at: %v: %w", key, err)
 	}
-	if err := b.Put(lastItemKey, newEnc); err != nil {
-		return fmt.Errorf("updating last item: %w", err)
+	lastItemIdx, err := s.lastItemIdx(ctx)
+	if err != nil && !errors.Is(err, leveldb.ErrNotFound) {
+		return err
 	}
-	if err := b.Put(countKey, []byte(strconv.Itoa(cnt+1))); err != nil {
-		return fmt.Errorf("updating length counter: %w", err)
+	if bytes.Compare(key, lastItemIdx) > 0 {
```

Comment:
I don't love assuming the return value of lastItemIdx is empty if leveldb.ErrNotFound is returned. In practice I'm sure it'll be empty, but I don't think it's good Go practice to read the value returned in an error is present.

Reply:
Done.